### DISCUSSION TITLE
SGS01 Temperature celsius vs fahrenheit

### DISCUSSION THREAD

**Discussion by Joao-Sousa-71** [make-all/tuya-local#1910] [Show and tell]
I've bought a SGS01 Soil moiture (BLE version) that works great with this integration however **I was having negative temperatures in HA**. I'm based in Europe and therefore the temperature units are set to Celsius.
The Smart Life mobile app was set also to Celsius, but SGS01 was reporting the tempertaure in fahrenheit. 
The way to fix the negative values was going to the Tuya Developper and in the device debugging do the following: Select the temp_unit_convert to "C" and press the send instruction bellow.

![image](https://github.com/make-all/tuya-local/assets/76400918/cde6ddb0-6897-46a6-a9ff-e594b932aa2b)

Hope this can help somebody.

**Comment by AdrianGarside** [comment]
I think I have a compatible device since it got detected as ble_smart_plant_moisture which is the config for the SGS01:
https://www.amazon.com/gp/product/B0CNKM3WPB/ref=ppx_yo_dt_b_asin_title_o04_s00?ie=UTF8&psc=1

In the SmartLife app it's ignoring my profile settings to use Fahrenheit and displaying in Centigrade. But HA is converting fine to Fahrenheit.

However, HA doesn't get any updated values unless I view the device in the SmartLife app. Then I see that apps cycle through historical values to present (seems it's collecting the recorded data off the device) and then both SmartLife and HA have updated values. Unfortunately, this means the HA integration isn't really useful if I have to keep running the SmartLife app anyway and also have to use that to get a complete historical graph. Are you seeing a different behavior?

**Reply by Joao-Sousa-71** [reply]
@AdrianGarside I have a similar behavior but we should be close to have that fixed. See this thread #1696 
Update: It is fixed.

**Comment by AdrianGarside** [comment]
Ok, so making the temperature unit selectable and setting it every 30 minutes via an automation triggers an update as a side effect?

**Reply by Joao-Sousa-71** [reply]
Yes, that is correct. Please use this last commit https://github.com/make-all/tuya-local/commit/44b1f5b197b9e861ef7f2c4cdd5b22806c228d01 that expose the temperature unit as a select sensor and then call the service in a schedule or on demand.

**Reply by AdrianGarside** [reply]
I think it's working. I'm finding the unit update is a bit unresponsive but it did seem to update the values without me opening the smart life app. It looks like it's necessary to toggle the setting which might be tricky with the lack of response I was seeing if I try and change it too often. HA seems to be backwards since I'm getting correct values only when the unit is set to celsius and HA is set to interpret the value as fahrenheit!?

**Reply by AdrianGarside** [reply]
Working ok-ish so far but I do have to flip the value back and forth and so the history graph has repetitive spikes down from the fahrenheit value to the celsius value each time I do it (hourly currently). Definitely way better than it was (value is actually being updated now) and the history graph is usable, just a little odd.

**Reply by AdrianGarside** [reply]
![image](https://github.com/make-all/tuya-local/assets/26147576/3e21d856-e264-4e5f-9e58-b4c8a4a3dc49)

**Reply by Joao-Sousa-71** [reply]
@AdrianGarside mine is working without flipping from Celsius to Fahrenheit… 
I’m sending to change to Celsius every 30 minutes. 
If you reduce too much the frequency the sensor do not react. Here are my charts from the last 24 hours:

![IMG_3266](https://github.com/make-all/tuya-local/assets/76400918/cc578591-1987-4e68-ae79-3c3356781ef4)

**Reply by AdrianGarside** [reply]
Hmm, I’ll try doing that in the automation - it didn’t seem to work through the UI.

I couldn’t get a select/set_option of c / f / Celsius / Fahrenheit to work - the automation gave an error that none of those were valid options. So I was setting firsf / last instead which worked and didn’t throw an error. Not sure why I was having that problem.

**Reply by make-all** [reply]
Try celsius or fahrenheit in all lower case

**Reply by AdrianGarside** [reply]
I thought I had tried that too but that worked. And while there no log of the action in home assistant it does seem to be triggering the device to update the temperature value in my single test just now.

**Reply by AdrianGarside** [reply]
Looking great now: 
![image](https://github.com/make-all/tuya-local/assets/26147576/5691b79d-4dfd-482d-8e51-b8d62a2b81d4)
